README.md

```diff
@@ -1,2 +1,308 @@
+</dependency>
+<dependency>
+    <groupId>com.h2database</groupId>
+    <artifactId>h2</artifactId>
+    <version>2.1.214</version>
+</dependency>
+```
+
+## Supported Databases
+* H2
+* PostgreSQL
+
+## How to use
+
+There are multiple steps you need to perform to correctly use Bibernate:
+
+1. Create `src/main/resources/persistence.yml` [resource file](#persistenceyml-file-example) with information about your
+   database.
+2. Create [entities](#entity-declaration-example), that represent table from the database.
+    * Mark them with the [`@Entity`](src/main/java/com/bobocode/bibernate/annotation/Entity.java) annotation.
+    * Add field that represents primary key of relevant table and annotated with
+      the [`@Id`](src/main/java/com/bobocode/bibernate/annotation/Id.java)
+3. Create [`SessionFactory`](src/main/java/com/bobocode/bibernate/session/SessionFactory.java) that would represent your
+   persistence unit using [`Persistence`](src/main/java/com/bobocode/bibernate/Persistence.java) class
+4. Create [`Session`](src/main/java/com/bobocode/bibernate/session/Session.java object to interact with a persistence context
+   using [`SessionFactory`](src/main/java/com/bobocode/bibernate/session/SessionFactory.java). Look at
+   the [example](#sessionsrcmainjavacombobocodebibernatesessionsessionjava-creation-example) below.
+
+#### `persistence.yml` file example:
+
+```yaml
+logLevel: TRACE
+persistenceUnit:
+  name: default
+  dataSource:
+    jdbcUrl: jdbc:h2:mem:testdb
+    user: sa
+    password: password
+  dialect: h2
+```
+
+#### Entity declaration example:
+
+```java
+import com.bobocode.bibernate.annotation.Column;
+import com.bobocode.bibernate.annotation.Entity;
+import com.bobocode.bibernate.annotation.Id;
+import com.bobocode.bibernate.annotation.Table;
+import lombok.Data;
+
+// Entity is required to be marked with @Entity
+@Entity
+public class Product {
+
+    // Entity is required to have public non-arg constructor
+    public Product() {
+    }
+
+    // Entity is required to have @Id field
+    @Id
+    private Long id;
+
+    private String name;
+
+    private Double price;
+    
+    // getters and setters are omitted for brevity
+}
+```
+
+#### [Session](src/main/java/com/bobocode/bibernate/session/Session.java) creation example:
+
+```java
+import com.bobocode.bibernate.Persistence;
+import com.bobocode.bibernate.session.Session;
+import com.bobocode.bibernate.session.SessionFactory;
+
+import java.util.Optional;
+
+public class SessionCreationExample {
+
+    public static void main(String[] args) {
+        SessionFactory sessionFactory = Persistence.createSessionFactory("default"); // creates session factory from persistence.yml properties
+        Session session = sessionFactory.openSession(); // opens Session that corresponds to DB Connection
+        try {
+            session.begin(); // starts transaction
+            Optional<T> product = session.find(Product.class, 1L); // finds product by ID
```

Comment:
Maybe here would be better to show example with save() method? as there is no any saved entities at the start of application.  (Yes it should not be our responsibility, but it can save time for a team who will review our project)